### DISCUSSION TITLE
Integrate prometheus client on a limited hardware capable server

### DISCUSSION THREAD

**Discussion by meetme2meat** [prometheus/client_golang#1239] [General]
More of an info need post: I'm aware of the Prometheus Golang client has been using it on a past Golang project. 
We want to implement the same in our existing project. but the system on which our application runs has very limited hardware. We are not sure the kind of load a Prometheus instrumentation would add to an existing running application.

Are there available benchmarks that can be referred to infer a number?

**Comment by bwplotka** [comment]
Let's move it to discussion, as there is no single answer here. It really depends on too many factors. How many metrics you have, their cardinality, if you use native metrics or collectors, how you implement collectors (if they are blocking, you make scrape more expensive too). 

What we can say is that this project is highly focused on efficiency, so if you feel there is a room for improvement we would be interested to optimize things. To my knowledge there is no faster/cheaper client for Prometheus metrics for Go in the public. There might be however rooms for improvement for certain edge cases.

**Reply by bwplotka** [reply]
Let's keep this discussion around so users can comment and share their benchmarks
